WebAPI/Controllers/StudentController.cs

```diff
@@ -83,6 +85,101 @@ public async Task<IActionResult> GetStudentTimetable(string userEmail)
             return Ok(timetable);
         }
 
+        [HttpGet("getStudentTimetableCalendar/{userEmail}")]
+        public async Task<IActionResult> GetStudentTimetableCalendar(string userEmail)
+        {
+            _logger.LogInformation($"[API getStudentTimetable] Getting timetable calendar for test student");
+            User user = await _userService.GetUserByEmailAsync(userEmail);
+            string studentId = user.Student.Id.ToString();
+            bool isValidGUID = Guid.TryParse(studentId, out Guid guid);
+            if (!isValidGUID)
+            {
+                return ErrorResponse($"Student id: {studentId} is not valid GUID.");
+            }
+
+            var student = await _studentService.FindByIdAsync(guid);
+            if (student == null)
+            {
+                return ErrorResponse($"Student with id: {studentId} does not exist.");
+            }
+
+            if (student.Timetable == null)
+            {
+                return ErrorResponse($"Timetable for student with id: {studentId} does not exist.");
+            }
+            DateTime dateNow = DateTime.Now;
+            StringBuilder sb = new StringBuilder();
+
+            sb.AppendLine("BEGIN:VCALENDAR");
+            sb.AppendLine("VERSION:2.0");
+            sb.AppendLine("CALSCALE:GREGORIAN");
+            sb.AppendLine("METHOD:PUBLISH");
+
+            sb.AppendLine("BEGIN:VTIMEZONE");
+            sb.AppendLine("TZID:Europe/Amsterdam");
+            sb.AppendLine("BEGIN:STANDARD");
+            sb.AppendLine("TZOFFSETTO:+0100");
+            sb.AppendLine("TZOFFSETFROM:+0100");
+            sb.AppendLine("END:STANDARD");
+            sb.AppendLine("END:VTIMEZONE");
+
+            var dayOfWeek = 0;
+            switch (dateNow.DayOfWeek)
+            {
+                case DayOfWeek.Sunday:
+                    dayOfWeek = 7;
+                    break;
+                case DayOfWeek.Monday:
+                    dayOfWeek = 1;
+                    break;
+                case DayOfWeek.Tuesday:
+                    dayOfWeek = 2;
+                    break;
+                case DayOfWeek.Wednesday:
+                    dayOfWeek = 3;
+                    break;
+                case DayOfWeek.Thursday:
+                    dayOfWeek = 4;
+                    break;
+                case DayOfWeek.Friday:
+                    dayOfWeek = 5;
+                    break;
+                case DayOfWeek.Saturday:
+                    dayOfWeek = 6;
+                    break;
+            }
```

Comment:
what about to replace whole switch with `dateNow.DayOfWeek == DayOfWeek.Sunday ? 7 : (int)dateNow.DayOfWeek`

Reply:
yeah, I was trying to retrieve the int value but did not try to cast it to int 😄